spec/section_spec.rb

```diff
@@ -49,6 +118,38 @@ class PageWithAnonymousSection < SitePrism::Page
   let(:locator) { instance_double('Capybara::Node::Element') }
   let(:section_with_block) { SitePrism::Section.new(Page.new, locator) { 1 + 1 } }
 
+  describe '#default_search_arguments' do
+    let(:base_section) { Class.new(SitePrism::Section) }
```

Comment:
I'd prefer a more legible integration-y style mock here. With a physical Page Object being instantiated through a more human interface.

Something like https://github.com/natritmeyer/site_prism/pull/250/files#diff-6fa85b9a8ee926d0d01e58459cddac2cR11 would be a good basis in my eyes

Reply:
I did it for the particular reason. Defining class integration-y will affect all unit tests and may potentially lead to false positive results. Because class will remain the same for all the test and redefinition/changes to this class will affect all the tests. For example,
```ruby
context '#hello' do
  class A
     def hello; 'Hello'; end
  end
  it 'should return hello' do
     expect(A.new.hello).to eql('Hello')
  end
end

context '#hi' do
  class A
     def hello; 'hi'; end
  end
  it { expect(A.new.hello).to eql('hi')}
end
```
in this case first example will fail

Reply:
Correct, due to overloading. However I'm in favour of doing some named examples scoping and having these classes reinstantiated with differing layouts (xpath / css), for example. And have a chunky PR reworking a load of specs for this.

Reply:
Ok, I'll change it